src/mds/FSMap.cc

```diff
@@ -341,7 +341,34 @@ void FSMap::get_health_checks(health_check_map_t *checks) const
   for (const auto &i : filesystems) {
     const auto &fs = i.second;
     health_check_map_t fschecks;
+
     fs->mds_map.get_health_checks(&fschecks);
+
+    // Some of the failed ranks might be transient (i.e. there are standbys
+    // ready to replace them).  We will report only on "stuck" failed, i.e.
+    // ranks which are failed and have no standby replacement available.
+    std::set<mds_rank_t> stuck_failed;
+
+    for (const auto &rank : fs->mds_map.failed) {
+      const mds_gid_t replacement = find_replacement_for(
+          {fs->fscid, rank}, {}, g_conf->mon_force_standby_active);
+      if (replacement == MDS_GID_NONE) {
+        stuck_failed.insert(rank);
+      }
+    }
+
+    // FS_WITH_FAILED_MDS
+    // MDS_FAILED
+    if (!stuck_failed.empty()) {
+      health_check_t& fscheck = checks->add(
```

Comment:
This fails with multifs because only one health warning of each type is allowed:

````
Assertion: /build/ceph-12.1.1-575-g7776c35/src/mon/health_check.h: 100: FAILED assert(checks.count(code) == 0)
ceph version 12.1.1-575-g7776c35 (7776c35bf2240350c9710c578bb27c65928ccee1) luminous (rc)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x10e) [0x7f77ea60f2ee]
 2: (()+0x2ce100) [0x7f77ea2bb100]
 3: (FSMap::get_health_checks(health_check_map_t*) const+0x2a9) [0x7f77ea7f08c9]
 4: (MDSMonitor::encode_pending(std::shared_ptr<MonitorDBStore::Transaction>)+0x7a6) [0x7f77ea5a2d26]
 5: (PaxosService::propose_pending()+0x2bd) [0x7f77ea506e5d]
 6: (C_MonContext::finish(int)+0x27) [0x7f77ea3c4647]
 7: (Context::complete(int)+0x9) [0x7f77ea3ffec9]
 8: (SafeTimer::timer_thread()+0xec) [0x7f77ea60bc9c]
 9: (SafeTimerThread::entry()+0xd) [0x7f77ea60d62d]
 10: (()+0x8184) [0x7f77e937d184]
 11: (clone()+0x6d) [0x7f77e7c3f37d]
3 jobs: ['1447958', '1448047', '1448063']
suites intersection: ['frag_enable.yaml', 'mount/fuse.yaml', 'overrides/{debug.yaml', 'whitelist_wrongly_marked_down.yaml}']
suites union: ['clusters/2-remote-clients.yaml', 'clusters/4-remote-clients.yaml', 'frag_enable.yaml', 'fs/basic_functional/{begin.yaml', 'fs/multifs/{begin.yaml', 'mon-debug.yaml', 'mount/fuse.yaml', 'no_client_pidfile.yaml', 'objectstore/bluestore.yaml', 'objectstore/filestore-xfs.yaml', 'overrides/{debug.yaml', 'tasks/data-scan.yaml}', 'tasks/failover.yaml}', 'whitelist_health.yaml', 'whitelist_wrongly_marked_down.yaml}']
````